doc/blog/201808-UnoUnderTheHoodButton.md

```diff
@@ -0,0 +1,145 @@
+      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" 
+      xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+      xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+      mc:Ignorable="d"> 
+
+    <StackPanel> 
+        <TextBlock x:Name="ClickTextBlock" 
+                   Text="Button wasn't clicked yet" /> 
+        <Button Content="Click me" 
+                Click="Button_Click" /> 
+    </StackPanel> 
+</Page> 
+````
+ 
+
+code-behind:
+
+```` csharp
+using System; 
+using Windows.UI.Xaml; 
+using Windows.UI.Xaml.Controls;  
+
+namespace UnoExtTestbed 
+{ 
+    public sealed partial class MainPage : Page 
+    { 
+        public MainPage() 
+        { 
+            this.InitializeComponent(); 
+        } 
+
+        private int _clickCount = 0; 
+        private void Button_Click(object sender, RoutedEventArgs e) 
+        { 
+            _clickCount++; 
+            ClickTextBlock.Text = $"Button was clicked {_clickCount} times."; 
+        } 
+    } 
+} 
+````
+ 
+
+I made a blank app using the [Uno Solution template](https://marketplace.visualstudio.com/items?itemName=nventivecorp.uno-platform-addin) and put this code on the main page. Whenever the [Button](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.button) is clicked, the number goes up. Add a bit more chrome and we could have a [viral hit](https://en.wikipedia.org/wiki/Cow_Clicker). 
+
+Note that the XAML is useful here but not obligatory, either on UWP or Uno. We could have defined and created all our views in C#, had we really wanted to.  
+## Exhibit A
+
+
+![Visualtree Uwp](Assets/Button/visualtree-uwp.png) ![Visualtree Android](Assets/Button/visualtree-android.png) ![Visualtree Ios](Assets/Button/visualtree-ios.png) ![Visualtree Wasm](Assets/Button/visualtree-wasm.png)
+*Visual tree information for UWP, Android, iOS, and WASM*
+
+So what does that get us? You can see the resulting visual trees on each target platform. They're all substantially similar. The top-level wrapping varies a little, but inside we can see the `MainPage`, `StackPanel`, `TextBlock` and `Button` we defined in XAML. (The `Frame` is created in the stock `App.xaml.cs` code that comes with the project template.)  
+
+You might notice there are a couple of extra views inside the `Button` that weren't explicitly defined in the XAML. These are part of Button's [default template](https://msdn.microsoft.com/en-us/library/windows/apps/mt299109.aspx?f=255&MSPPError=-2147217396). If you're not familiar with UWP/WPF/Silverlight/etc's concept of templating, there's a lot to say about [the subject](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/control-templates), but the gist of it is that any view which inherits from `Control` is a tabula rasa, an empty vessel which will be filled with subviews as defined in its template. (Some of these subviews might be templated controls themselves.) 
```

Comment:
Remove the error params from the default template link

Reply:
>  will be filled with subviews 

_inner elements_ instead ? or _inner UIElements_ ? subviews is probably too UIKit specific.

Reply:
Good point